_posts/2024-11-18-debezium-3-0-2-final-released.adoc

```diff
@@ -0,0 +1,241 @@
+THis configuration is historically used to control how the Oracle connector behaves when a DDL failure is observed.
+
+In this release, the `event.processing.failure.handling.mode` is also used to control failures for DML-based events.
+If there was an issue with the Oracle connector parsing your _insert_, _update_, or _delete_ operations, you can safely configure the connector to _fail_, _warn_, or _skip_ the DML event based on your needs (https://issues.redhat.com/browse/DBZ-8208[DBZ-8208]).
+
+[NOTE]
+====
+The default behavior is to always _fail_ when an event is not safely handled by the connector.
+By adjusting this to _warn_ or _skip_, while the connector will safely continue past the failed event, you will introduce data loss and will needs to be addressed manually.
+====
+
+=== Vitess
+
+==== Performance improvements
+
+In earlier builds of the Debezium for Vitess connector, the connector used a regular expression-based filter system that matches all tables based on a prefix with varying suffixes, and later exclusions would be applied based on configuration.
+This has the potential to waste CPU and create hotspots due to creating intermediate objects for the event that would later to be filtered and garbage collected.
+
+In this release, we've improved the way the Vitess connector processes this use case by applying the filtration earlier in the event processing chain.
+This should reduce the number of intermediate objects created and improve the overall performance of the connector.
+For key spaces that have the same prefix and differing suffixes, this should provide better overall performance than older builds (https://issues.redhat.com/browse/DBZ-8354[DBZ-8354]).
+
+=== Sink connectors
+
+Debezium 0.x introduced a common source-connector framework that has become the foundation for source connectors provided by the project, including our community-led connectors such as Spanner, Vitess, and others.
+With the introduction of the MongoDB sink connector recently, our long-term goal is to approach sink connectors in a similar way, providing a common sink-connector framework to ease the creation of Debezium-based sink connectors.
+
+Over Debezium 3.x lifecycle, you will see incremental steps to streamline the source across the JDBC and MongoDB sink connectors.
+We will minimize disruptions in maintenance releases as you have come to expect, but expect future major and minor releases to introduce deprecations and changes to support this endeavor.
+
+In this first round of changes, we've introduced a new Debezium module, `debezium-sink`.
+This module acts as the foundation for all sink connectors and is home to a variety of common classes, including the `SinkConnectorConfig` class, naming strategy implementations, and the common representation of a sink record, `DebeziumSinkRecord`.
```

Comment:
I don't necessarily agree with this since we have other code snippets in the same sentence, so I'll leave it as written.  I did address the other one.